src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -706,6 +722,41 @@ namespace OpenLoco::Vehicles
         carComponent.body->updateCargoSprite();
     }
 
+    // 0x004AF4D6
+    void insertCar(VehicleBogie& source, VehicleBase& dest)
+    {
+        if (source.id == dest.id)
+        {
+            return;
+        }
+        Ui::WindowManager::invalidate(Ui::WindowType::vehicle, enumValue(source.head));
+        Ui::WindowManager::invalidate(Ui::WindowType::vehicle, enumValue(dest.getHead()));
+        Ui::WindowManager::invalidate(Ui::WindowType::vehicleList);
+
+        Vehicle sourceTrain(source.head);
+        auto precedingVehicleComponent = source.previousVehicleComponent();
+        for (auto& car : sourceTrain.cars)
+        {
+            if (car.front->id != source.id)
+            {
+                continue;
+            }
+            auto& lastBody = car.body;
```

Comment:
this should be a pointer `auto* lastBody = &car.body;` the reason is that L750 currently copies instead of points to what you want it to be.